chalice/app.py

```diff
@@ -450,7 +516,11 @@ def __call__(self, event, context):
                                                     function_args)
         if self._cors_enabled_for_route(route_entry):
             self._add_cors_headers(response, route_entry.cors)
-        return response.to_dict()
+        if self._matches_content_type(response.case_insensitive_headers.get(
```

Comment:
Yes, the underlying function is tested, but your use case is not given this is new code.  We're testing that "the content type value returned by a user can have '; extra-stuff' in it and still match" regardless of however the underlying functionality, `_matches_content_type` has been tested.